vector/src/main/java/im/vector/util/DinsicUtils.java

```diff
@@ -210,4 +226,119 @@ public void onClick(DialogInterface dialog, int id) {
 
     }
 
+    //=============================================================================================
+    // Handle existing direct chat room
+    //=============================================================================================
+
+    /**
+     * Open the current direct chat with the corresponding user id.
+     * Look for a potential existing direct chat with this user (by considering pending invite too)
+     *
+     * @param participantId : participant id (matrix id ou email)
+     * @param canCreate create the direct chat if it does not exist.
+     * @param mActivity current activity
+     * @param mSession current session
+     * @param mSpinnerView
+     * @param mCreateDirectMessageCallBack
+     * @return boolean that says if the direct chat room is opened or not
+     */
+    public static boolean openDirectChat(String participantId, boolean canCreate, final Activity mActivity, final MXSession mSession, final View mSpinnerView, ApiCallback mCreateDirectMessageCallBack) {
```

Comment:
Don't you think the activity should be the first param?

I would suggest the following header with some changes in the parameters order:
* Select the most suitable direct chat for the provided user identifier (matrix or third party id).
* During this search, the pending invite are considered too.
* The selected room (if any) may be opened synchronously or not. It depends if the room is ready or not.
*
* @param mActivity current activity
* @param participantId : participant id (matrix id ou email)
* @param mSession current session
* @param canCreate create the direct chat if it does not exist.
* @param mCreateDirectMessageCallBack, the ApiCallback used only in case of room creation
* @param mSpinnerView spinner displayed during the asynchronous actions.
* @return boolean that says if a direct chat room has been selected or not.